wemake_python_styleguide/visitors/base.py

```diff
@@ -271,3 +276,45 @@ def run(self) -> None:
         for token in self.file_tokens:
             self.visit(token)
         self._post_visit()
+
+
+class BaseCSTVisitor(BaseVisitor, CSTVisitor, metaclass=abc.ABCMeta):
+    """
+    Allows to store violations while traversing node cst.
+
+    This class should be used as a base class for all ``cst`` based checkers.
+    Method ``visit()`` is defined in ``CSTNode`` class.
+
+    Attributes:
+        tree: ``libcst.Module`` tree to be checked.
+
+    """
+
+    METADATA_DEPENDENCIES = (PositionProvider,)  # noqa: WPS115
```

Comment:
Can you please document and type this attribute?

Reply:
done